Sources/TLS/Socket.swift

```diff
@@ -17,20 +18,31 @@ public final class Stream {
          - parameter context: Re-usable SSL.Context in either Client or Server mode
          - parameter descriptor: The file descriptor from an unsecure socket already created.
     */
-    public init(context: Context, certificates: Certificates, socket: Int32) throws {
+    public init(context: Context, certificates: Certificates, config: TLSConfig, socket: Int32) throws {
         self.context = context
+        self.config = config
         cConfig = tls_config_new()
         self.socket = socket
 
         self.certificates = certificates
```

Comment:
I think we could move certificates into config. This would match the libressl more closely I think since it has two high level objects `config` and `ctx`. So the swift equivalents would be `Config` and `Context`. 

Based on this line: 
```swift
tls_configure(context.cContext, cConfig)
```

I think that `Config` should take a `Context`. Also `tls_init()` (called in Context) needs to be created before any other methods can be called. So requiring a context be created as the first step is critical.